packages/module-utils/test/generate-dts.spec.ts

```diff
@@ -215,6 +216,61 @@ describe('Generate DTS', function () {
         expect(tk.typecheck('test.ts')).to.include(propUnknownNotOnType);
     });
 
+    it('should not expose imported symbols', () => {
+        tk.populate({
+            'origin.st.css': `
+                .cls {
+                    container-name: cont;
+                }
+                @property --customProp;
+                :vars {
+                    buildVar: red;
+                }
+                @keyframes anim {}
+                @layer layer1 {}
+            `,
+            'test.st.css': `
+                @st-import CompRoot, [
+                    cls,
+                    --customProp,
+                    buildVar,
+                    keyframes(anim),
+                    container(cont),
+                    layer(layer1),
+                ] from './origin.st.css';
+                .cls { color: value(buildVar); }
+                .CompRoot { 
+                    color: green;
+                    --customProp: green;
+                    animation: anim;
+                }
+                @container cont (inline-size > 1px) {}
+                @layer layer1 {}
+            `,
+            'test.ts': `
+                import { eq } from "./test-kit";
+                import { classes, vars, stVars, keyframes, containers, layers } from "./test.st.css";
+                
+                eq<string>(classes.cls);
+                eq<string>(classes.CompRoot);
+                eq<string>(vars.customProp);
+                eq<string>(vars.customProp);
+                eq<string>(stVars.buildVar);
+                eq<string>(keyframes.anim);
+                eq<string>(containers.cont);
+                eq<string>(layers.layer1);
+            `,
+        });
+
+        expect(tk.typecheck('test.ts')).to.include(propNotOnType('cls'));
```

Comment:
How does this does checks that something is not exported?

Reply:
The test import symbols from `origin` stylesheet to `test` stylesheet and then checks that d.ts that describes the runtime doesn't contain any leaked imported symbols.

The actual runtime maps are checked as part of the core. I might have skipped over testing the runtime e2e - will add!